pkg/sql/catalog/multiregion/region_config.go

```diff
@@ -468,3 +519,109 @@ func CanDropRegion(name catpb.RegionName, config RegionConfig) error {
 	}
 	return CanSatisfySurvivalGoal(config.survivalGoal, len(config.regions)-1)
 }
+
+// getHomeRegionConstraintConjunction returns the ConstraintsConjunction from
+// a list of ConstraintsConjunction for the given region name.
+// If we found the constraint for the region, we return the corresponding
+// ConstraintsConjunction and "true" as the second boolean returned value.
+// If not found in the list, it returns an empty lease preference and "false".
+func getHomeRegionConstraintConjunction(
+	constraints []zonepb.ConstraintsConjunction, homeRegion string,
+) (constraintConjunction zonepb.ConstraintsConjunction, found bool) {
+	for _, constraint := range constraints {
+		// TODO(janexing): We should also consider the case with contraints on zone
+		// level.
+		if constraint.Constraints[0].Key == "region" &&
+			constraint.Constraints[0].Value == homeRegion {
+			return constraint, true
+		}
+	}
+	return zonepb.ConstraintsConjunction{}, false
+}
+
+// getHomeRegionLeasePreference returns the lease preference and its index from
+// a list of lease preference for the given region name. If not found in the
+// list, it returns an empty lease preference and -1 as the index.
+func getHomeRegionLeasePreference(
+	leasePreferences []zonepb.LeasePreference, homeRegion string,
+) (zonepb.LeasePreference, int) {
+	for idx, leasePreference := range leasePreferences {
+		// TODO(janexing): We should also consider the case with contraints on zone
+		// level.
+		if leasePreference.Constraints[0].Value == homeRegion {
+			return leasePreference, idx
+		}
+	}
+	return zonepb.LeasePreference{}, -1
+}
+
+// validateZoneConfigExtension ensures that the zone config extensions don't
+// set configs that disagree with their home region.
+func validateZoneConfigExtension(
+	ext zonepb.ZoneConfig, zc zonepb.ZoneConfig, homeRegion string,
+) error {
+	if homeRegion == "" {
+		return nil
+	}
+
+	// If the extension does not inherit that field from the original zone config,
+	// we will use the one in the extension, and hence we need to validate that
+	// field.
+	// Lease preferences.
+	if !(ext.ShouldInheritLeasePreferences(&zc)) && len(zc.LeasePreferences) > 0 && len(ext.LeasePreferences) > 0 {
+		oriHomeRegionLeasePreference, idx1 := getHomeRegionLeasePreference(zc.LeasePreferences, homeRegion)
```

Comment:
Rather we just use `originalHomeRegionLeasePreference` took me a second to understand what `ori` meant

Reply:
Done.